configs/default/SPRO-SPRACINGH7EXTREME.config

```diff
@@ -0,0 +1,225 @@
+# Betaflight / STM32H750 (SP7E) 4.4.0 Dec 26 2022 / 12:28:11 (7671c7eb01) MSP API: 1.45
+
+#define TARGET_BOARD_IDENTIFIER "SP7E"
+#define USBD_PRODUCT_STRING "SPRacingH7EXTREME"
+
+#
+# Settings required to boot
+#
+# Note: hardware for all possible config storage subsystems needs to be known and initialised BEFORE loading
+# a config from it can work. (SPI/QSPI/SDCARD).
+#
+
+#define EEPROM_SIZE=8192
+
+#define USE_SPRACING_PERSISTENT_RTC_WORKAROUND
+
+#define USE_BUTTONS
+#define BUTTON_A_PIN            PE4
+#define BUTTON_A_PIN_INVERTED // Active high
+#define BUTTON_B_PIN            PE4
+#define BUTTON_B_PIN_INVERTED // Active high
+
+#define USE_QUADSPI
+#define USE_QUADSPI_DEVICE_1
+#define QUADSPI1_SCK_PIN PB2
+#define QUADSPI1_BK1_IO0_PIN PD11
+#define QUADSPI1_BK1_IO1_PIN PD12
+#define QUADSPI1_BK1_IO2_PIN PE2
+#define QUADSPI1_BK1_IO3_PIN PD13
+#define QUADSPI1_BK1_CS_PIN PB10
+#define QUADSPI1_BK2_IO0_PIN PE7
+#define QUADSPI1_BK2_IO1_PIN PE8
+#define QUADSPI1_BK2_IO2_PIN PE9
+#define QUADSPI1_BK2_IO3_PIN PE10
+#define QUADSPI1_BK2_CS_PIN NONE
+#define QUADSPI1_MODE QUADSPI_MODE_BK1_ONLY
+#define QUADSPI1_CS_FLAGS (QUADSPI_BK1_CS_HARDWARE | QUADSPI_BK2_CS_NONE | QUADSPI_CS_MODE_LINKED)
+
+#define USE_SPI
+
+#define USE_SPI_DEVICE_2
+#define SPI2_SCK_PIN            PD3
+#define SPI2_MISO_PIN           PC2
+#define SPI2_MOSI_PIN           PC3
+#define SPI2_NSS_PIN            PB12
+
+#define USE_SPI_DEVICE_3
+#define SPI3_SCK_PIN            PB3
+#define SPI3_MISO_PIN           PB4
+#define SPI3_MOSI_PIN           PD6
+#define SPI3_NSS_PIN            PA15
+
+#define USE_SPI_DEVICE_4
+#define SPI4_SCK_PIN            PE12
+#define SPI4_MISO_PIN           PE13
+#define SPI4_MOSI_PIN           PE14
+#define SPI4_NSS_PIN            PE11
+
+#define USE_FLASH_TOOLS
+#define USE_FLASH_W25N01G
+#define FLASH_QUADSPI_INSTANCE    QUADSPI
+#define CONFIG_IN_EXTERNAL_FLASH
+
+#define USE_SDCARD
+#define SDCARD_DETECT_PIN PD10
+#define SDCARD_DETECT_INVERTED
+#define SDIO_DEVICE             SDIODEV_1
+#define SDIO_USE_4BIT           true
+#define SDIO_CK_PIN             PC12
+#define SDIO_CMD_PIN            PD2
+#define SDIO_D0_PIN             PC8
+#define SDIO_D1_PIN             PC9
+#define SDIO_D2_PIN             PC10
+#define SDIO_D3_PIN             PC11
+
+#define TARGET_IO_PORTA 0xffff
+#define TARGET_IO_PORTB 0xffff
+#define TARGET_IO_PORTC 0xffff
+#define TARGET_IO_PORTD 0xffff
+#define TARGET_IO_PORTE 0xffff
+#define TARGET_IO_PORTF 0xffff
+#define TARGET_IO_PORTG 0xffff
+
+#
+# Settings to enable/configure remaining hardware
+#
+
+#define USE_USB_ID
+
+#define USE_I2C
+#define USE_I2C_DEVICE_1
+#define I2C1_SCL PB8
+#define I2C1_SDA PB9
+#define I2C_DEVICE (I2CDEV_1)
+
+#define _USE_MAG
+#define _USE_MAG_HMC5883
+#define _USE_MAG_QMC5883
+
+#define _USE_BARO
+#define _USE_BARO_BMP388
+
+#define USE_ACC
+#define USE_ACC_SPI_MPU6500
+#define USE_GYRO
+#define USE_GYRO_SPI_MPU6500
+#define ENSURE_MPU_DATA_READY_IS_LOW
+
+#define USE_PID_AUDIO
```

Comment:
This is a user option. Can be added to the options a user can add in their build.

Reply:
it's not a user option, the audio hardware needs to be initialised.  enabling the pid audio FEATURE however IS a user option.

Reply:
Ok for clarity this should be a user Build option, ie do I want to include this feature at all or not. Whether I run time include it is a different story. Of course it is your target config so if you want to force users to include it... happy days.

Reply:
no. the pin connected to the audio amp must be configured as ANALOG_OUT.  At all times.  It is not acceptable and bad for the on-board audio hardware if the pin for instance is set to any other GPIO mode.   It NEEDs to be present in the build.  Uses should not be able to map any other function to the pin used by pid audio.

The actual code size difference for enabling PID audio is quite small.